lib/natalie/compiler/macro_expander.rb

```diff
@@ -73,7 +79,17 @@ def macro?(node)
             end
           end
         elsif node.sexp_type == :iter
-          macro?(node[1])
+          get_macro_name(node[1])
+        else
+          get_hidden_macro_name(node)
+        end
+      end
+
+      # "Hidden macros" are just regular-looking Ruby code we intercept at compile-time.
+      # We will try to support common Ruby idioms here that cannot be done at runtime.
+      def get_hidden_macro_name(node)
+        if node[0..2] == s(:call, s(:gvar, :$LOAD_PATH), :<<)
```

Comment:
I'm not familiar with Natalie's internals. Would this also catch `$: <<`?

Reply:
No, but I added a commit to support that one too. Thanks!